### ISSUE TITLE
Decision Reviews: Add new issue heading announced multiple times

### ISSUE THREAD

**Issue by davidakennedy** [department-of-veterans-affairs/va.gov-team#72521]
## Value Statement

As a screen reader user adding a new issue on the notice of disagreement form
I want to be able to hear the step title only once
So that I can move onto adding the issue without confusion.

## Background Context

When adding a new issue on the Notice of Disagreement form, VoiceOver on MacOS will announce the heading, "Add an issue" three times. We're managing focus properly, and the markup looks good. More research needs to be done to determine the cause for this. Some possibilities:

- It's nested in a `<fieldset>` in a `<legend>`.

I didn't experience this elsewhere in the form. I also don't think it's a blocker to launch because the content is being announced and the Veteran isn't blocked from moving forward.

### GIF

![Add new issue link clicked and focus moved to "Add new issue" heading](https://github.com/department-of-veterans-affairs/va.gov-team/assets/1473618/a58a4b38-c115-46a3-8531-5693ab6f2d25)

### To reproduce

1. Turn on VoiceOver with Safari on MacOS
2. Proceed to the "Select the issues you’d like us to review" page
3. Activate the "Add new issue" link
4. Listen for the announcement of the heading three times

## Acceptance Criteria

TBD

## Tasks

TBD

## Definition of Ready 

TBD

## Definition of Done

TBD

**Comment by saderagsdale** [comment]
@davidakennedy will check on the root cause of this issue w/ other accessibility specialists.

**Comment by HeatherWidmont** [comment]
@davidakennedy is still researching & we will discuss estimate when he has a clearer picture of a path forward

**Comment by davidakennedy** [comment]
@HeatherWidmont Apologies, had planned to post this yesterday, but lost time because of the power outage.

After some more in-depth testing, here's what I found:

- This does happen because a heading is nested inside of a `<legend>` element. VoiceOver appears to be reading the heading out once because it's focused on, once for the legend content and once for the grouped content. At least that's my working theory. I can't find any existing bugs with Apple/WebKit/VoiceOver around this, but Apple doesn't have a great history of making that public. This could be just the way it works.
- This currently only happens with VoiceOver on MacOS, not on iOS. This doesn't happen in any of the other major screen reader and/or browser combinations, including: JAWS, NVDA, TalkBack or Narrator.
- If I change VoiceOver's mode to navigate by group instead of by DOM order, the heading gets read out twice vs. three times.

Recommendations:

- Since this is only happening in VoiceOver on MacOS, I don't think we should prioritize a fix above other important work. This seems like either a VoiceOver bug, which is likely given the behavior isn't consistent on iOS, or intentional.
- For now, I would leave this be, and potentially explore a different approach down the road if Apple doesn't make a fix here. My rationale is that while this is annoying, it doesn't happen everywhere and on every screen reader. Plus, it doesn't block someone from moving forward.
- As far as how to fix it if we wanted to do that, I'd recommend leaving the headings inside the legend, but instead of shifting focus to the heading, we could shift focus to the legend element itself. To do that we would need to add make it focusable via `tabindex="-1"` first, then shift focus.

I [created a demo to help test this](https://cdpn.io/pen/debug/YzgWqGN/e85bb8d7c75495c86f8fe4e1fd71e6f4), and prove out a possible solution. My recommended solution is the second demo in there.

A good question for engineers (cc: @rocio-desantiago): What's the level of effort on the recommended fix? Maybe across the NOD form, and then potentially others.

**Comment by rocio-desantiago** [comment]
@davidakennedy assuming it's only a tab index update, a low level of effort but beyond that could require more research.

**Comment by HeatherWidmont** [comment]
@rocio-desantiago to investigate a little bit more & then estimate this ticket

**Comment by rocio-desantiago** [comment]
Tested locally and in staging. Hear it repeating those mentioned headers twice. 
Tab-index='-1' didn't make a difference. 
@davidakennedy can you confirm if you're still hearing it be read out loud 3 times? I wonder if our VoiceOver's are different versions.

**Comment by rocio-desantiago** [comment]
**Sharing screenshots from Codepen (Chrome) with unexpected behavior.
Will follow up with DK when he returns on 1/24.**

| 1st button |
| ----- |
| <img width="1196" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWZ5QWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--2efbb1b3ba738f4a2973b6e7be8a7e704a4256b7/button1-h3.png" alt="button1-h3.png" /> |


| 2nd and 3rd button |
| ----- |
| <img width="1246" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWJ5QWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--522915c9d28b4b81a38721dffa4c47c883a80945/addAnIssue-group.png" alt="addAnIssue-group.png" /> |

**Comment by rocio-desantiago** [comment]
**Update:** Tested in Safari and do see the expect behavior. 
Above is what we see in Chrome.

> Do we happen to have stats on what browsers and what screenreaders veterans use to access VA.gov?

**Comment by davidakennedy** [comment]
> Do we happen to have stats on what browsers and what screenreaders veterans use to access VA.gov?

Checking in on this the stats question.

**Comment by rubensun** [comment]
@rocio-desantiago looking into fixing on the safari end, will check in with @davidakennedy

**Comment by davidakennedy** [comment]
![View of Google Analytics showing about 45 percent of traffic comes from mobile.](https://github.com/department-of-veterans-affairs/va.gov-team/assets/1473618/918639b0-b01d-42e9-965c-e23bccf59183)

![View of Google Analytics 47 percent of sessions are mobile; 60 percent of those mobile sessions come from iOS devices.](https://github.com/department-of-veterans-affairs/va.gov-team/assets/1473618/0a5018ca-d840-49d1-a339-95d486807952)

Both of these views show the last 30 days or so.

From this, more than half of our mobile traffic comes from iOS devices. There isn't a way to know how many of these folks use VoiceOver on iOS.

From all the testing I've done on this, I'm okay leaving it as is. My reasoning here is because this only happens in VoiceOver, and I'd rather us not get into coding around particular implementations in certain screen readers.

**Comment by rocio-desantiago** [comment]
Resolved that it doesn't make sense to fix for only one of the screenreaders. This ticket is good to be closed.

**Comment by HeatherWidmont** [comment]
@rocio-desantiago to add estimate using the story estimator

**Comment by rocio-desantiago** [comment]
Updated estimate
